### ISSUE TITLE
t/03_compiles_bad.t fails after upgrading IPC::Run to 0.048 on Windows

### ISSUE THREAD

**Issue by plicease** [plicease/Test-Script#1]
See originally reported rt bug:

https://rt.cpan.org/Ticket/Display.html?id=94685

I am able to reproduce it with Strawberry 5.20.1:

```
straw-N:\home\ollisg\dev\Test-Script>prove -lv t\03_compiles_bad.t
prove -lv t\03_compiles_bad.t
Inappropriate I/O control operation at N:\home\ollisg\dev\Test-Script\lib/Test/Script.pm line 117.
t\03_compiles_bad.t .. 
1..5
ok 1 - Found bad script
Dubious, test returned 25 (wstat 6400, 0x1900)
Failed 4/5 subtests 

Test Summary Report
-------------------
t\03_compiles_bad.t (Wstat: 6400 Tests: 1 Failed: 0)
  Non-zero exit status: 25
  Parse errors: Bad plan.  You planned 5 tests but ran 1.
Files=1, Tests=1,  0 wallclock secs ( 0.03 usr +  0.00 sys =  0.03 CPU)
Result: FAIL

```

**Comment by plicease** [comment]
This is the likely culprit:

https://metacpan.org/diff/file?target=RJBS/IPC-Run3-0.047-TRIAL/lib/IPC/Run3.pm&source=RJBS/IPC-Run3-0.046/lib/IPC/Run3.pm

**Comment by plicease** [comment]
This actually appears to be a bug in IPC::Run3 as reported here:

https://rt.cpan.org/Ticket/Display.html?id=95308

The proposed patch 

```patch
         $sys_call_time = gettimeofday() if profiling;
 
+        $! = 0;                  # make sure we don't test below against some previous error
+
         my $r = ref $cmd
               ? system { $cmd->[0] } is_win32 ? quote_native( @$cmd ) : @$cmd
               : system $cmd;
@@ -416,7 +418,7 @@
 
         if (
             defined $r
-            && ( $r == -1 || ( is_win32 && $r == 0xFF00 ) )
+            && ( $r == -1 || ( is_win32 && $r == 0xFF00 && $errno != 0 ) )
             && !$options->{return_if_system_error}
         ) {
             croak( $errno );
```

makes this test pass.

Also related:

https://rt.cpan.org/Ticket/Display.html?id=46333

**Comment by nanis** [comment]
I can verify that the proposed patch also fixes this error on 64-bit Windows 8.1 using `perl` 5.20.2 compiled with Visual Studio 2013 tools. [rjbs/IPC-Run3#9](https://github.com/rjbs/IPC-Run3/issues/9).

**Comment by plicease** [comment]
It seems this will never be fixed in `IPC::Run3`.  We have a workaround, or can use `Capture::Tiny`.
